src/main/kotlin/com/github/pemistahl/lingua/api/LanguageDetector.kt

```diff
@@ -259,8 +260,9 @@ class LanguageDetector internal constructor(
                 totalLanguageCounts.incrementCounter(UNKNOWN)
             } else if (wordLanguageCounts.size == 1) {
                 val language = wordLanguageCounts.toList().first().first
+                var wordSize = if (language in Constant.LANGUAGES_SPLIT_BY_NO_SPACE) wordLanguageCounts[language] else 1
                 if (language in languages) {
-                    totalLanguageCounts.incrementCounter(language)
+                    wordSize?.let { totalLanguageCounts.incrementCounter(language, it) }
```

Comment:
1. I think the block you put your code in is the wrong place here. This only works for input strings consisting of one and only one language. What if the input string contains substrings in both English and Chinese, for example? Then your algorithm would not work as intended. You actually have to iterate over all languages in `wordLanguageCounts` and check whether they belong to the subset supporting logograms.

2. A Chinese substring consisting of five characters, for example, without any spaces in between the characters would be counted as a single word. Therefore, `wordLanguageCounts` would contain the count 1 for this language. However, in line 263, you would have to determine the length of the Chinese substring instead of looking into the map `wordLanguageCounts`. So, I'm afraid, your whole approach in solving this problem ought to be reworked a bit.

3. In line 263, there is no need to make `wordSize` mutable using `var`. Please make it immutable using `val` instead.

4. In line 265, you check whether `wordSize` is null or not. However, `wordSize` can never be null as defined in line 263. So I think it would be safe to just write `totalLanguageCounts.incrementCounter(language, wordSize)`.

Reply:
Yep, the previous word would be the basic unit to count size(The size only be 1). The better way to solve the problem is that correct the method of ```LanguageDetector.splitTextIntoWords```. If every word belong to one language, the problem will be sovled.If i want to correct the ```LanguageDetector.splitTextIntoWords``` , the text will be scaned all characters in ```LanguageDetector.splitTextIntoWords```. And in ```LanguageDetector.detectLanguageWithRules``` also  scan all characters. And It cost double time to do the same things( scaning all characters). That's why i put code  here. That's the problem, I will find the way to solve this. Thanks again, next time i will consider your advices before i push the code.